source/base/partitioner.cc

```diff
@@ -58,6 +58,39 @@ namespace Utilities
 
 
 
+    Partitioner::Partitioner(const types::global_dof_index local_size,
+                             const types::global_dof_index ghost_size,
+                             const MPI_Comm                communicator)
+      : global_size(Utilities::MPI::sum<types::global_dof_index>(local_size,
+                                                                 communicator))
+      , locally_owned_range_data(global_size)
+      , local_range_data(std::pair<types::global_dof_index,
+                                   types::global_dof_index>(0, local_size))
```

Comment:
Do you want to use `[prefix_sum, prefix_sum + local_size)` here as well?

Reply:
It is overridden later.

Reply:
I would prefer to set `local_range_data` in this function already to avoid an invalid state; all I see here is `locally_owned_range_data` but `local_range_data != locally_owned_range_data` (even though one of the two is redundant). But we could postpone it to another PR in case this function is reworked again.

Reply:
You are right. I got confused by the similar names. What about the version now?